integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java

```diff
@@ -196,8 +197,9 @@ public void tearDown() {
   // Since Operator version 3.1.0 the service pod prefix has been changed 
   // from -external to -ext e.g.
   // domain1-adminserver-ext  NodePort    10.96.46.242   30001:30001/TCP 
-  private void upgradeOperator(String domainType, String operatorVersion, String externalServiceNameSuffix,
-                               boolean useHelmUpgrade) {
+  private void upgradeOperator(String domainType, String operatorVersion, 
```

Comment:
It seems there is a lot going on in this method. Can you move the domain creation into another method and call it from here. Also, you can cleanup some code, for older versions of operator like 2.5.0 we uninstall and install operator again for upgrade. We don't have upgrade tests for those releases now, that code can be removed. Lastly change the method name to installAndUpgradeOperator or something else which describes what this method is doing.

Reply:
Removed the unused code  (a) creating domain from yaml file (b) operator uninstall/install process

Reply:
domain creation code is still in upgradeOperator, it can be refactored for better readability.

Reply:
Consolidated the domain creation code to a single method in latest update.

Reply:
renamed the method  upgradeOperator --> installAndUpgradeOperator
Added utility methods to the main method as follows  ..
{
....
// install operator with older release 
    HelmParams opHelmParams = installOperator(operatorVersion,
                 opNamespace, domainNamespace);
    // create WLS domain and verify
    createWlsDomainAndVerify(domainType, domainNamespace, domainVersion,
           externalServiceNameSuffix);
    // upgrade to latest operator
    upgradeOperatorAndVerify(externalServiceNameSuffix,
          opNamespace, domainNamespace);
}